src/types/errors/selling-partner-api-errors.ts

```diff
@@ -47,11 +47,14 @@ export class SellingPartnerNotFoundError extends SellingPartnerGenericError {}
 export class SellingPartnerRequestTooLongError extends SellingPartnerGenericError {}
 export class SellingPartnerUnsupportedMediaTypeError extends SellingPartnerGenericError {}
 export class SellingPartnerTooManyRequestsError extends SellingPartnerGenericError {
-  public rateLimit?: string
+  public rateLimit?: number
 
   public constructor(error: ModelError, headers: Headers) {
     super(error, headers)
-    this.rateLimit = headers['x-amzn-RateLimit-Limit'] || headers['x-amzn-ratelimit-limit']
+    this.rateLimit =
+      Number(headers['x-amzn-RateLimit-Limit']) ||
```

Comment:
We need to guard here against `NaN`. I recommend to create a small helper function to deal with this.

`Number(undefined)` = `NaN`
`Number("foo")` = `NaN`

We certainly don't want any NaNs :)

Reply:
> We certainly don't want any NaNs :)

```
NaN || NaN || undefined // undefined
```

@moltar : 
logical OR (||) operator against `NaN` already? 🤔 
I don't think we need a helper function.

Reply:
Good point 😁 👍🏼